docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_encrypted_remediation.mdx

```diff
@@ -1,6 +1,218 @@
+   - Choose a KMS key from the dropdown or create a new one if necessary.
+
+4. **Review and Launch:**
+   - Review all the configurations to ensure encryption is enabled.
+   - Click on the "Create cluster" button to launch the cluster with encryption enabled.
+
+By following these steps, you ensure that your Redshift clusters are encrypted, thereby preventing the misconfiguration of having unencrypted clusters.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To ensure that your Amazon Redshift clusters are encrypted using the AWS CLI, you can follow these steps:
+
+1. **Create a KMS Key**:
+   First, you need to create a KMS (Key Management Service) key that will be used to encrypt your Redshift clusters.
+   ```sh
+   aws kms create-key --description "Redshift encryption key"
+   ```
+
+2. **Create a Redshift Cluster with Encryption**:
+   When creating a new Redshift cluster, specify the KMS key to enable encryption.
+   ```sh
+   aws redshift create-cluster \
+       --cluster-identifier my-redshift-cluster \
+       --node-type dc2.large \
+       --master-username myuser \
+       --master-user-password mypassword \
+       --cluster-type single-node \
+       --kms-key-id <KMS_KEY_ARN>
+   ```
+
+3. **Modify an Existing Redshift Cluster to Enable Encryption**:
+   If you have an existing Redshift cluster, you need to take a snapshot, then restore the cluster from the snapshot with encryption enabled.
+   ```sh
+   aws redshift create-cluster-snapshot \
+       --cluster-identifier my-redshift-cluster \
+       --snapshot-identifier my-snapshot
+   ```
+
+   ```sh
+   aws redshift restore-from-cluster-snapshot \
+       --snapshot-identifier my-snapshot \
+       --cluster-identifier my-new-redshift-cluster \
+       --kms-key-id <KMS_KEY_ARN>
+   ```
+
+4. **Verify Encryption**:
+   Finally, verify that the cluster is encrypted by describing the cluster and checking the `Encrypted` field.
+   ```sh
+   aws redshift describe-clusters \
+       --cluster-identifier my-new-redshift-cluster \
+       --query 'Clusters[*].{ClusterIdentifier:ClusterIdentifier,Encrypted:Encrypted}'
+   ```
+
+By following these steps, you can ensure that your Amazon Redshift clusters are encrypted, thereby preventing the misconfiguration of having unencrypted clusters.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Redshift Clusters from being unencrypted in AWS Redshift using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to ensure that your Redshift clusters are encrypted:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Set up a Boto3 session with the necessary AWS credentials and region.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>